src/Symfony/Component/Cache/Adapter/TraceableAdapter.php

```diff
@@ -0,0 +1,210 @@
+        $result = $call->result;
+        $call->isHit = $result->isHit();
+
+        // Display the result in a good way depending on the data type
+        if ($call->isHit) {
+            $call->result = $this->getValueRepresentation($result);
+        } else {
+            $call->result = null;
+        }
+
+        $this->calls[] = $call;
+
+        return $result;
+    }
+
+    public function hasItem($key)
+    {
+        $call = $this->timeCall(__FUNCTION__, array($key));
+        $this->calls[] = $call;
+
+        return $call->result;
+    }
+
+    public function deleteItem($key)
+    {
+        $call = $this->timeCall(__FUNCTION__, array($key));
+        $this->calls[] = $call;
+
+        return $call->result;
+    }
+
+    public function save(CacheItemInterface $item)
+    {
+        $arg = $this->getValueRepresentation($item);
+
+        $call = $this->timeCall(__FUNCTION__, array($item));
+        $call->arguments = array($arg);
+        $this->calls[] = $call;
+
+        return $call->result;
+    }
+
+    public function saveDeferred(CacheItemInterface $item)
+    {
+        $arg = $this->getValueRepresentation($item);
+
+        $call = $this->timeCall(__FUNCTION__, array($item));
+        $call->arguments = array($arg);
+        $this->calls[] = $call;
+
+        return $call->result;
+    }
+
+    public function getItems(array $keys = array())
+    {
+        $call = $this->timeCall(__FUNCTION__, array($keys));
+        $result = $call->result;
+
+        $f = function() use ($result, $call) {
+            $hits = 0;
+            $items = array();
+            foreach ($result as $item) {
+                $items[] = $item;
+                if ($item->isHit()) {
+                    ++$hits;
+                }
+
+                $call->result = $this->getValueRepresentation($items);
+                $call->hits = $hits;
+                $call->count = count($items);
+
+                yield $item;
+            }
+        };
+
+        $call->result = 'NULL';
+        $call->hits = 0;
+        $call->count = 0;
+        $this->calls[] = $call;
+
+        return $f();
+    }
+
+    public function clear()
+    {
+        $call = $this->timeCall(__FUNCTION__, array());
+        $this->calls[] = $call;
+
+        return $call->result;
+    }
+
+    public function deleteItems(array $keys)
+    {
+        $call = $this->timeCall(__FUNCTION__, array($keys));
+        $this->calls[] = $call;
+
+        return $call->result;
+    }
+
+    public function commit()
+    {
+        $call = $this->timeCall(__FUNCTION__);
+        $this->calls[] = $call;
+
+        return $call->result;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getCalls()
+    {
+        return $this->calls;
+    }
+
+    /**
+     * Get a string to represent the value.
+     *
+     * @param mixed $value
+     *
+     * @return string
+     */
+    private function getValueRepresentation($value)
```

Comment:
I think we don't need this logic, let's just use `$value` directly.
The clone may fail (implem can throw in `__clone`), and is fragile (not recursive).
and for other types of data, we don't care (resources cannot come up from a cache pool)
removing also will make the layer slimmer, which is also a nice target